OneCommunityGlobal/HGNRest
# Initial commit

Taking a look at the recent reviews. Kindly do not review the PR unless you see a different kind of bug to report



# Description
- A “Limit See-All” option for teams 

-  Other Links ->Teams

- When this option is turned on for a team member,  the team member  shouldn't  be able to see  other members of the team in their dashboard 

- The member however should be able to see team members of other teams  they are part of 

## Related PRS (if any):
This backend PR is related to the [1905](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1905) frontend PR.
To test this backend PR you need to checkout the [1905](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1905) frontend PR.

## Main changes explained:

- Added  a "visible" attribute to Member Object in Teams document and set it to true by default.
- Added a new route to updating the team member visibility. 
- Add  a  function  called  "updateTeamVisibility "in teams Controller   to process the newly selected visibility of the team member


## How to test:
1. check into current branch
2. run node src/utilities/addMembersToTeams.js so that "visible" attribute gets added to all existing team members
_(This is expected to take time. This step is not needed if you are testing a new team with new team members)_
4. run `npm install ` followed by `npm run build start`  and  `npm start ` to run this PR locally
5.  Now follow the steps mentioned in the front end PR


## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/89441435/887ed506-2ce6-443d-beb4-645abc471335


## Note:
- Kindly mention any  bugs , errors you see in the console. 
- Try breaking the functionality , any suggestion is helpful
